internal/db/collection_retriever.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package db
+
+import (
+	"context"
+
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/internal/db/description"
+)
+
+// CollectionRetriever is a helper struct that retrieves a collection from a document ID.
+type CollectionRetriever struct {
```

Comment:
suggestion: This type doesn't need to be public. It's the internal db implementation of `kms.CollectionRetriever` interface. It's not meant to be used directly.

Reply:
done